src/helics/network/tcp/TcpCommsSS.cpp

```diff
@@ -151,6 +151,8 @@ void TcpCommsSS::queue_tx_function()
     }
     gmlc::networking::TcpServer::pointer server;
     auto ioctx = gmlc::networking::AsioContextManager::getContextPointer();
+    auto sf = encrypted ? gmlc::networking::SocketFactory(encryption_config) :
```

Comment:
The downside of reading the encryption config file here is that a user could theoretically change the contents of the config file before it gets used to create a socket.

Reply:
Even if they did, not sure why that would be a problem.